src/platform_impl/windows/icon.rs

```diff
@@ -239,13 +233,26 @@ impl WinCursor {
                 return Err(io::Error::last_os_error());
             }
 
-            Ok(Self::from_handle(handle))
+            Ok(Self::Cursor(Arc::new(RaiiCursor { handle })))
+        }
+    }
+
+    pub(crate) fn build<T>(
+        cursor: OnlyCursorImageBuilder,
+        _: &EventLoopWindowTarget<T>,
+    ) -> WinCursor {
+        match WinCursor::new(&cursor.0) {
+            Ok(cursor) => cursor,
+            Err(err) => {
+                warn!("Failed to create custom cursor: {err}");
+                WinCursor::Failed
```

Comment:
Seeing this, I think the API should actually be something like:
```rust
impl CustomCursor {
    pub fn from_rgba(
        rgba: impl Into<Vec<u8>>,
        width: u16,
        height: u16,
        hotspot_x: u16,
        hotspot_y: u16,
    ) -> CustomCursorBuilder;
}

impl CustomCursorBuilder {
    pub fn build<T>(self, window_target: &EventLoopWindowTarget<T>) -> Result<CustomCursor, BadImage>;
}
```

But that can be done later.

Reply:
Agreed, will do that in a follow-up.